File: MODULO3/PYTHON/TRABAJANDO_PYTHON/asociacion_muchos_a_muchos_estudiante_curso/inscripcion.py


class Inscripcion:
    def __init__(self, estudiante=None, curso=None, fecha_inscripcion=""):
        self.estudiante = estudiante
        self.curso = curso
        self.fecha_inscripcion = fecha_inscripcion


    @property   
    def estudiante(self):
        return self._estudiante
    @estudiante.setter
    def estudiante(self, value):
        self._estudiante = value

    @property
    def curso(self):
        return self._curso
    @curso.setter
    def curso(self, value):
        self._curso = value
    
    @property   
    def fecha_inscripcion(self):
        return self._fecha_inscripcion
    @fecha_inscripcion.setter
    def fecha_inscripcion(self, value):
        self._fecha_inscripcion = value
           
    def __str__(self):
        return f"Inscripción de {self.estudiante} en el curso {self.curso} fecha_inscripción: {self.fecha_inscripcion}"
    

File: MODULO3/UML/DiagramaUML/src/asociacion_tipo_muchos_a_muchos_estudiante_curso/Inscripcion.java
package asociacion_tipo_muchos_a_muchos_estudiante_curso;

public class Inscripcion {

    // VARIABLES DE INSTANCIA
    private Estudiante estudiante;
    private Curso curso;
    private String fechaInscripcion;

    // CONSTRUCTORES
    public Inscripcion() {
        this.estudiante = null;
        this.curso = null;
        this.fechaInscripcion = "";
    }

    public Inscripcion(Estudiante estudiante, Curso curso, String fechaInscripcion) {
        this.estudiante = estudiante;
        this.curso = curso;
        this.fechaInscripcion = fechaInscripcion;
    }
    
    // METODOS SET Y GET
    public Estudiante getEstudiante() {
        return estudiante;
    }

    public void setEstudiante(Estudiante estudiante) {
        this.estudiante = estudiante;
    }

    public Curso getCurso() {
        return curso;
    }

    public void setCurso(Curso curso) {
        this.curso = curso;
    }

    public String getFechaInscripcion() {
        return fechaInscripcion;
    }

    public void setFechaInscripcion(String fechaInscripcion) {
        this.fechaInscripcion = fechaInscripcion;
    }
    
    // METODO TOSTRING

    @Override
    public String toString() {
        return "Inscripcion{" + "estudiante=" + estudiante + ", curso=" + curso + ", fechaInscripcion=" + fechaInscripcion + '}';
    }
}
